pylint/lint.py

```diff
@@ -1342,15 +1358,46 @@ def _worker_check_single_file(file_item):
     _worker_linter.open()
     _worker_linter.check_single_file(name, filepath, modname)
 
+    mapreduce_data = collections.defaultdict(list)
+    for checker in _worker_linter.get_checkers():
+        try:
+            data = checker.get_map_data()
+        except AttributeError:
```

Comment:
Do only the `Similar` checker have a `get_map_data` function ? Couldn't it be also on the `BaseChecker` so every Checker have it ?

Reply:
Your call, it makes sense either way. The intent was to respect existing APIs and separate out concerns, but having it on base-checker might be super useful :)

Reply:
However it seems that some Checkers can be launched on each file of a code base in parallel without work (for example Docstring checker only need the code of a single file to analyze it) but the Similar checker need to have information about the whole code base to find similarity. I'm under the impression that get_map_data is useful for the Similar checker but would not be required to parallelize come other checkers correctly. Maybe I did not understand the purpose of the mapreduce you did ? What do you think ?

Reply:
That is correct, for now only a single Checker needs this functionality.

Reply:
OK, let's kee it simple then.